### ISSUE TITLE
OData Client takes significant amount of time to load and parse large EDM model

### ISSUE THREAD

**Issue by danleydmello** [OData/odata.net#2205]
I have a very large EDM model, more than 2500 entities and some of these entities have a large number of properties and relationships.
I am building Blazor Web assembly app using the Odata client. It takes about 45 secs to load and parse the model on the client.
 
           ```
 using (var stream = await httpClient.GetStreamAsync("$metadata"))
            using (var reader = XmlReader.Create(stream))
            {
                _edmModel = CsdlReader.Parse(reader);
            }
```
Is there a way I can speed it up or skip the model entirely? The UI is very simple and only needs few entities from the model. 

Thanks in advance

**Comment by marabooy** [comment]
Hi @danleydmello, You could try and generate a client using[ connected service](https://marketplace.visualstudio.com/items?itemName=marketplace.ODataConnectedService) this should cache the model if the model does not change alot and also in the generation step it should allow you to generate code for entities that you are going to use.

**Comment by danleydmello** [comment]
@marabooy , Thanks for the reply. I tried that method already but I think because there are so many classes in the model the auto generation process crash. May be I should try with 64 bit VS 2022 :)

**Comment by marabooy** [comment]
Let me also check what other options are available in the meantime do try using vs2022.

**Comment by danleydmello** [comment]
ok thats great Thanks @marabooy

**Comment by sfalik** [comment]
I am facing this same issue.  the [Connected Service](https://marketplace.visualstudio.com/items?itemName=marketplace.ODataConnectedService) extension crashes VS2019, and it's not yet supported in VS2022

**Comment by xuzhg** [comment]
@sfalik CS can be used in VS2022.

**Comment by sfalik** [comment]
hmmm, @xuzhg it only seems to list 2017 and 2019 -- do I need to pull a different version of it?

![image](https://user-images.githubusercontent.com/17600811/136064212-d867761e-879d-40af-b1ee-ceb3fdad5331.png)

**Comment by xuzhg** [comment]
@sfalik Did you try it using 2022?

**Comment by sfalik** [comment]
@xuzhg yes, it won't install to VS2022.  And if you try to search for it from within VS2022 extension market it doesn't show up.

**Comment by xuzhg** [comment]
Ok. I see. https://github.com/OData/ODataConnectedService/issues/225
